snapcraft/plugins/v1/catkin.py

```diff
@@ -853,7 +866,10 @@ def _find_system_dependencies(catkin_packages, rosdep, catkin):
         # let's get the dependencies for the entire workspace.
         dependencies |= rosdep.get_dependencies()
 
-    for dependency in dependencies:
+    # Remove skip keys
+    filtered_dependencies = [x for x in dependencies if x not in skip_keys]
```

Comment:
I suggest loading the skip-keys into a set so you can do this far more efficiently.

Reply:
Should be fixed.